spring-xd-module/src/main/java/org/springframework/xd/module/spark/streaming/SparkStreamingDriverModule.java

```diff
@@ -273,8 +321,14 @@ public void stop() {
 		logger.info("stopping SparkDriver");
 		try {
 			// todo: when possible (spark 1.3.0), change to streamingContext.stop(false, true) without the cancel
-			streamingContext.ssc().sc().cancelAllJobs();
-			streamingContext.close();
+			if (javaStreamingContext != null) {
+				javaStreamingContext.ssc().sc().cancelAllJobs();
+				javaStreamingContext.close();
+			}
+			if (streamingContext != null) {
+				streamingContext.sc().cancelAllJobs();
+				streamingContext.stop(true, false);
```

Comment:
but do we want to stop the underlying SparkContext itself? I'm mainly concerned that we are not consistent here (for the Java version, we just call close())